kart/gui/mapswipetool.py

```diff
@@ -56,6 +56,11 @@ def canvasReleaseEvent(self, e):
         self.canvas().setCursor(QCursor(Qt.PointingHandCursor))
 
     def canvasMoveEvent(self, e):
+        if (e.x(), e.y()) == (self.firstPoint.x(), self.firstPoint.y()):
+            # the moveEvent was fired immediately after the press event,
+            # don't change the swipe direction
+            return
+
```

Comment:
I wonder if it's better to move this further down?

I think the issue is that dX=0; dY=0, then isVertical is set to False always, even if it was True before? Whereas maybe we should `if not dX and not dY: return` at that point?

Reply:
It also seems to happen depending on where the first click is and you can end up with a dX > dY. I'm not totally clear on all the conditions where it happened but it was at least some times where the values were non-zero and it got stuck as True. The case that I ran into this was that the swipe would always turn vertical and stay that way.

Reply:
On helper.py, yes will need the rebase, just depends on the order that these might be applied.

Reply:
I merged the other 2 PRs